### DISCUSSION TITLE
Unable to set a timeout duration for analytics session

### DISCUSSION THREAD

**Discussion by vijayst** [invertase/react-native-firebase#4405] [Q&A] [answered]
I am using `firebase.analytics().setTimeoutDuration(300000)` to set a analytics session timeout of 5 minutes. But analytics session expires only after 30 minutes. In firebase debug view, both ga_session_id and ga_session_number remain the same even after 5 minutes of inactivity. Any idea why setTimeoutDuration is not working for my app?

**Comment by mikehardy** [comment, marked as answer]
Hi there! I'm unsure - we haven't had any reports of this not working, but no reports of it working either - so my standard suggestion here is focused on self-diagnosis, I'd follow these steps.

- make sure you are on current versions of everything (including overriding the firebase-ios-sdk and firebase-android-sdk to current versions of 6.34.0 and 25.12.0 respectively - https://rnfirebase.io/#overriding-native-sdk-versions)
- go right in to node_modules in the javascript and native calls for the affected platform and instrument them (simple console.log / System.out.println etc) to make sure they APIs are being called with arguments you expect and everything is hooked up to the underlying SDK
- if RNFB is doing everything correctly down to the firebase SDK level, then check their support tickets (they are on github too) and if necessary do a quickstart to reproduce the problem for them for attention: https://github.com/firebase/quickstart-android/tree/master/analytics is an example, they have ios too)
- if RNFB is not doing everything correctly, open a ticket here

Hope this helps
